trafix-codec/src/message/mod.rs

```diff
@@ -0,0 +1,193 @@
+//! Implementation of the message module.
+
+pub mod field;
+
+use crate::message::field::{
+    Field,
+    value::{begin_string::BeginString, msg_type::MsgType},
+};
+
+/// Represents the header section of a FIX message.
+///
+/// The header always contains the protocol [`BeginString`] (tag 8)
+/// and the message type [`MsgType`] (tag 35), and may include
+/// additional session or routing fields.
+pub struct Header {
+    /// The `BeginString` identifying the FIX protocol version.
+    #[allow(dead_code)]
+    begin_string: BeginString,
+
+    /// The `MsgType` indicating the business purpose of the message (message type).
+    #[allow(dead_code)]
+    msg_type: MsgType,
+
+    /// Optional additional header fields.
+    fields: Option<Vec<Field>>,
```

Comment:
I think we can spare us the `Option` here and use empty `Vec<Field>` to indicate absent fields.